doc/releases/changelog-dev.md

```diff
@@ -4,6 +4,43 @@
 
 <h3>New features since last release</h3>
 
+* Leveraging quantum just-in-time compilation to optimize parameterized hybrid workflows with the momentum
+  quantum natural gradient optimizer is now possible with the new :class:`~.MomentumQNGOptimizerQJIT` optimizer.
+  [(#7606)](https://github.com/PennyLaneAI/pennylane/pull/7606)
+
+  Similar to the :class:`~.QNGOptimizerQJIT` optimizer, :class:`~.MomentumQNGOptimizerQJIT` offers a 
+  `jax.jit`- and `qml.qjit`-compatible analogue to the existing :class:`~.MomentumQNGOptimizer` with an 
+  Optax-like interface:
+
+  ```python
+  import pennylane as qml
+  import jax.numpy as jnp
+
+  @qml.qjit(autograph=True)
```

Comment:
Once again a bit confused by the choice to `qjit` the whole workflow. I feel like it'd be better to showcase being able to qjit individual steps. If the user forgets the `autograph=True` part, this will have *substantial* performance penalties.  It seems like we are recommending something that they are likely to mess up. I'd prefer to recommend the more robust syntax.

Reply:
Same comment on the class docstring too.  I'd recommend showcasing jitting/ qjitting individual steps instead.

Reply:
I briefly discussed about this also with @Alex-Preciado and the recommended strategy here would be:

1. After the code example showing how to use `qjit` on the whole workflow, add to the docstring also an example to show its usage on a single optimization step (as we did in the `QNGOptimizerQJIT` docstring)
2. For both `QNGOptimizerQJIT` and `MomentumQNGOptimizerQJIT`, add a more exhaustive disclaimer/note pointing out the necessity of enabling `autograph` and a short explanation of why this is the case

Reply:
Just because it's how they will doesn't mean it's how they should. And in our examples we can provide what they should to influence how they will.

Reply:
@SimoneGasperini yours and Alex's suggestion seems like a good plan 👍

Reply:
Ok thank you @isaacdevlugt 👌🏽

I've just updated the disclaimer both for QNG and Momentum-QNG. Take a look at [45e6f3d](https://github.com/PennyLaneAI/pennylane/pull/7606/commits/45e6f3d916c9910d70a24bf2c3b48c674f75db93) and feel free to change it.